libs/debugging/tests/unit/print.cpp

```diff
@@ -0,0 +1,129 @@
+//  Copyright (c) 2020 John Biddiscombe
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+// The purpose of this test is to ensure that when enable_print is false
+// that function arguments and other parameters used in the print statements
+// are completely elided
+
+#include <hpx/debugging/print.hpp>
+#include <hpx/threading_base/print.hpp>
+#include <hpx/util/high_resolution_timer.hpp>
+#include <hpx/modules/testing.hpp>
+
+#include <string>
+#include <chrono>
+
+namespace hpx {
+    // this is an enabled debug object that should output messages
+    static hpx::debug::enable_print<true>  p_enabled("PRINT  ");
+    // this is disabled and we want it to have zero footprint
+    static hpx::debug::enable_print<false> p_disabled("PRINT  ");
+}   // namespace hpx
+
+int increment(std::atomic<int> &counter) {
+    return ++counter;
+}
+
+#define AB_LAZY(expr, debug) \
```

Comment:
Is this is a typo? The code below uses `DB_LAZY`.

Reply:
Sorry. Left over from an experiment. Removed